xtractmime/_patterns.py

```diff
@@ -1,102 +1,156 @@
+"""
+Section 6.1, step 1
+https://mimesniff.spec.whatwg.org/commit-snapshots
+/609a3a3c935fbb805b46cf3d90768d695a1dcff2/#matching-an-image-type-pattern
+"""
```

Comment:
Please, use a comment. Module-level docstrings are meant to document the module itself.

If these module-level constants where to go into Sphinx documentation, you could use `#:` instead of `#` for the comments, and Sphinx would be able to pick up the comments as documentation for the following module-level object. But since that’s not the case here, a simple comment should suffice.

Also, [it’s unusual to split URLs, even if they surpass the line length guideline](https://stackoverflow.com/a/10740010/939364). I think static analyzers (i.e. black and flake8) don’t complain in these cases; but it they do, it’s better to add an inline flag to ignore these specific errors than to split URLs. This applies to `_utils.py` as well.